dist/secure-handlebars.js

```diff
@@ -9346,24 +9364,38 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, stateObj, charNo) {
                 this._charNo = node.content.index+1;
             }
         }
+
+        return parser;
     }
-    analyzeAstTree.call(this, ast.left, 0);
-    analyzeAstTree.call(this, ast.right, 1);
 
 
     /* make lastStates[0] and lastStates[1] the same as the tree has one branch */
-    ast.left.length > 0 && ast.right.length === 0? r.lastStates[1] = r.lastStates[0] : '';
-    ast.left.length === 0 && ast.right.length > 0? r.lastStates[0] = r.lastStates[1] : '';
-    debug("analyzeAst:["+r.lastStates[0].state+"/"+r.lastStates[1].state+"]");
+    // ast.left.length > 0 && ast.right.length === 0? r.lastStates[1] = r.lastStates[0] : '';
+    // ast.left.length === 0 && ast.right.length > 0? r.lastStates[0] = r.lastStates[1] : '';
+    // debug("analyzeAst:["+r.lastStates[0].state+"/"+r.lastStates[1].state+"]");
 
     // if the two branches result in different state
-    if (r.lastStates[0].state !== r.lastStates[1].state) {
```

Comment:
as discussed, we need to enhance the state comparison under the principle that for those internal states affecting the future decision of customized filters added should be in the deepCompareState function